File: apps/askbible-android/app/src/main/java/me/askbible/native_/ui/PlanWidgets.kt
package me.askbible.native_.ui

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.RowScope
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.asPaddingValues
import androidx.compose.foundation.layout.navigationBars
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyListScope
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import me.askbible.native_.data.Brand
import me.askbible.native_.data.Parchment
import me.askbible.native_.data.PlanCopy
import me.askbible.native_.data.PlanFact

/**
 * 读经计划页的手机版视觉件（两端同构，iOS 见 Read/PlanWidgets.swift）。
 * Josh 2026-09-09：介绍与内容要为手机看优化 —— 字不小于 13，段落改成要点，能点的做成卡片，整体简短、可视。
 * 文案来自 data/bible-reading-plans/mobile-brief.zh-CN.json（PlanText = PlanCopy 的 mobile.* 键）。
 */
object PlanText {
    fun t(key: String): String = PlanCopy.t("mobile.$key")
    fun f(key: String, args: Map<String, String>): String = PlanCopy.f("mobile.$key", args)
}

/** mobile-brief 里的图标名 → Material 字形 */
fun planIconGlyph(name: String): String = when (name) {
    "today" -> MI.TODAY
    "loop" -> MI.LOOP
    "swap" -> MI.SWAP_HORIZ
    "stairs" -> MI.STAIRS
    "repeat" -> MI.REPEAT
    "book" -> MI.MENU_BOOK
    "sync" -> MI.SYNC
    "spa" -> MI.SPA
    "replay" -> MI.REPLAY
    "calendar" -> MI.CALENDAR_MONTH
    "layers" -> MI.LAYERS
    else -> MI.TODAY
}

/** 轨道图标与色：旧约蓝 / 新约橙 / 智慧书绿（目录页的约别配色） */
fun planTrackStyle(id: String): Pair<String, Color> = when (id) {
    "ot" -> Pair(MI.HISTORY_EDU, Color(0xFF2E5E8C))
    "nt" -> Pair(MI.AUTO_STORIES, Color(0xFFB8611E))
    else -> Pair(MI.LIGHTBULB, Color(0xFF3F7A4A))
}

/** 子页左上返回键（RN ShellSystemBackButton），与 iOS 的 PlanBackButton 对等 */
@Composable
fun PlanBackButton(onBack: () -> Unit, theme: Parchment = Parchment.light) {
    Box(Modifier.statusBarsPadding().padding(start = 12.dp, top = 8.dp).size(44.dp).clickableNoRipple(onBack), contentAlignment = Alignment.Center) {
        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = null, tint = theme.ink.toColor(), modifier = Modifier.size(26.dp))
    }
}

/** 羊皮卷子页的滚动容器：顶部留出返回键；独立页没有底栏，底部只留导航条 + 24 */
@Composable
fun PlanPageColumn(theme: Parchment = Parchment.light, content: LazyListScope.() -> Unit) {
    val navBottom = WindowInsets.navigationBars.asPaddingValues().calculateBottomPadding()
    Box(Modifier.fillMaxSize()) {
        ParchmentBackground(theme = theme)
        LazyColumn(
            Modifier.fillMaxSize().statusBarsPadding(),
            contentPadding = PaddingValues(start = 20.dp, end = 20.dp, top = 64.dp, bottom = navBottom + 24.dp),
            content = content,
        )
    }
}

/** 下划线小链接 */
@Composable
fun PlanLink(text: String, theme: Parchment = Parchment.light, size: Int = 14, color: Color = theme.muted.toColor(), modifier: Modifier = Modifier, onClick: () -> Unit) {
    Text(text, modifier.clickableNoRipple(onClick), color = color, fontSize = size.sp, textDecoration = TextDecoration.Underline)
}

/** 要点胶囊：图标 16 + 文字 14/600 */
@Composable
fun PlanFactChip(fact: PlanFact, theme: Parchment = Parchment.light) {
    Row(
        Modifier.clip(CircleShape).background(theme.surface.toColor().copy(alpha = 0.85f)).border(0.5.dp, theme.border.toColor(), CircleShape)
            .padding(horizontal = 12.dp, vertical = 7.dp),
        verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        MaterialIcon(planIconGlyph(fact.icon), 18f, theme.muted.toColor())
        Text(fact.text, color = theme.ink.toColor(), fontSize = 15.sp, fontWeight = FontWeight.SemiBold)
    }
}

/** 要点 chips 一行，放不下就折行（卡片里三颗 chip 曾被挤得每颗都换行） */
@OptIn(ExperimentalLayoutApi::class)
@Composable
fun PlanFactRow(facts: List<PlanFact>, centered: Boolean = false, theme: Parchment = Parchment.light) {
    FlowRow(
        Modifier.fillMaxWidth(),
        horizontalArrangement = if (centered) Arrangement.spacedBy(8.dp, Alignment.CenterHorizontally) else Arrangement.spacedBy(8.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        for (f in facts) PlanFactChip(f, theme)
    }
}

/** 「怎么读」一行：金色圆底图标 + 16/500 短句 */
@Composable
fun PlanHowRow(fact: PlanFact, theme: Parchment = Parchment.light) {
    Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(12.dp)) {
        Box(Modifier.size(36.dp).clip(CircleShape).background(Brand.logo.toColor().copy(alpha = 0.2f)), contentAlignment = Alignment.Center) {
            MaterialIcon(planIconGlyph(fact.icon), 20f, theme.ink.toColor())
        }
        Text(fact.text, Modifier.weight(1f), color = theme.inkSoft.toColor(), fontSize = 17.sp, lineHeight = 25.sp, fontWeight = FontWeight.Medium)
    }
}

/** 今日读经卡：轨道图标 / 轨道名 13 / 书章 22/700 / 说明 14 / 可选进度条；整卡可点进章 */
@Composable
fun PlanTodayCard(track: String?, label: String, title: String, subtitle: String?, progress: Double? = null, theme: Parchment = Parchment.light, onClick: () -> Unit) {
    Row(
        Modifier.fillMaxWidth().clip(RoundedCornerShape(14.dp)).background(Color(0xB8FFFCF5)).border(0.5.dp, theme.border.toColor(), RoundedCornerShape(14.dp))
            .clickableNoRipple(onClick).padding(horizontal = 16.dp, vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(14.dp),
    ) {
        if (track != null) {
            val (glyph, color) = planTrackStyle(track)
            Box(Modifier.size(46.dp).clip(CircleShape).background(color.copy(alpha = 0.14f)), contentAlignment = Alignment.Center) {
                MaterialIcon(glyph, 24f, color)
            }
        }
        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(label, color = theme.faint.toColor(), fontSize = 14.sp, fontWeight = FontWeight.SemiBold, letterSpacing = 0.6.sp)
            Text(title, color = theme.ink.toColor(), fontSize = 22.sp, lineHeight = 28.sp, fontWeight = FontWeight.Bold, maxLines = 2, overflow = TextOverflow.Ellipsis)
            if (subtitle != null) Text(subtitle, color = theme.muted.toColor(), fontSize = 15.sp)
            if (progress != null) PlanProgressBar(progress, Modifier.padding(top = 4.dp), theme)
        }
        MaterialIcon(MI.CHEVRON_RIGHT, 26f, theme.faint.toColor())
    }
}

@Composable
fun PlanProgressBar(value: Double, modifier: Modifier = Modifier, theme: Parchment = Parchment.light) {
    Box(modifier.fillMaxWidth().height(5.dp).clip(CircleShape).background(theme.border.toColor())) {
        Box(Modifier.fillMaxWidth(value.coerceIn(0.0, 1.0).toFloat()).height(5.dp).clip(CircleShape).background(Brand.logo.toColor()))
    }
}

/** 整宽主按钮 52 高：filled = 墨底米字；否则米底墨字（次要） */
@Composable
fun PlanPrimaryButton(title: String, filled: Boolean = true, theme: Parchment = Parchment.light, onClick: () -> Unit) {
    Box(
        Modifier.fillMaxWidth().height(52.dp).clip(RoundedCornerShape(14.dp))
            .background(if (filled) theme.ink.toColor() else theme.surface.toColor())
            .border(1.dp, if (filled) Color.Transparent else theme.ink.toColor().copy(alpha = 0.35f), RoundedCornerShape(14.dp))
            .clickableNoRipple(onClick),
        contentAlignment = Alignment.Center,
    ) {
        Text(title, color = if (filled) Color(0xFFF5EFE4) else theme.ink.toColor(), fontSize = 17.sp, fontWeight = FontWeight.SemiBold)
    }
}

@Composable
fun PlanSectionHeader(title: String, hint: String? = null, theme: Parchment = Parchment.light) {
    Column(Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(4.dp)) {
        Text(title, color = theme.ink.toColor(), fontSize = 20.sp, fontWeight = FontWeight.Bold)
        if (hint != null) Text(hint, color = theme.muted.toColor(), fontSize = 15.sp)
    }
}

/** 「✓ 当前计划 · 第 N 天」金色胶囊 */
@Composable
fun PlanStatusPill(text: String) {
    Row(
        Modifier.clip(CircleShape).background(Brand.logo.toColor().copy(alpha = 0.26f)).padding(horizontal = 12.dp, vertical = 7.dp),
        verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        MaterialIcon(MI.CHECK_CIRCLE, 18f, Color(0xFF8A5A00))
        Text(text, color = Color(0xFF5B3A00), fontSize = 15.sp, fontWeight = FontWeight.SemiBold)
    }
}

/** 单选格（深读节奏 / 起算方式）：标题 18/700 + 副题 13，选中金底金边 */
@Composable
fun RowScope.PlanChoiceTile(title: String, subtitle: String, on: Boolean, theme: Parchment = Parchment.light, onClick: () -> Unit) {
    Column(
        Modifier.weight(1f).clip(RoundedCornerShape(12.dp))
            .background(if (on) Brand.logo.toColor().copy(alpha = 0.28f) else theme.surface.toColor().copy(alpha = 0.6f))
            .border(if (on) 1.5.dp else 0.5.dp, if (on) Brand.logo.toColor() else theme.border.toColor(), RoundedCornerShape(12.dp))
            .clickableNoRipple(onClick).padding(vertical = 12.dp, horizontal = 8.dp),
        horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Text(title, color = theme.ink.toColor(), fontSize = 18.sp, fontWeight = FontWeight.Bold)
        Text(subtitle, color = theme.muted.toColor(), fontSize = 14.sp, fontWeight = FontWeight.Medium, textAlign = TextAlign.Center)
    }
}

/** 第几天步进器：44 触控键 + 24/700 数字 */
@Composable
fun PlanStepper(value: Int, min: Int, max: Int, theme: Parchment = Parchment.light, onChange: (Int) -> Unit) {
    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(16.dp)) {
        PlanStepButton(MI.REMOVE, value > min, theme) { onChange(maxOf(min, value - 1)) }
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(4.dp)) {
            Text("第", color = theme.muted.toColor(), fontSize = 16.sp)
            Text("$value", Modifier.width(44.dp), color = theme.ink.toColor(), fontSize = 24.sp, fontWeight = FontWeight.Bold, textAlign = TextAlign.Center)
            Text("天", color = theme.muted.toColor(), fontSize = 16.sp)
        }
        PlanStepButton(MI.ADD, value < max, theme) { onChange(minOf(max, value + 1)) }
    }
}

@Composable
private fun PlanStepButton(glyph: String, enabled: Boolean, theme: Parchment, onClick: () -> Unit) {
    Box(
        Modifier.size(44.dp).alpha(if (enabled) 1f else 0.4f).clip(RoundedCornerShape(10.dp)).background(theme.surface.toColor())
            .border(0.5.dp, theme.border.toColor(), RoundedCornerShape(10.dp)).clickableNoRipple { if (enabled) onClick() },
        contentAlignment = Alignment.Center,
    ) { MaterialIcon(glyph, 22f, theme.ink.toColor()) }
}

/** 展开 / 收起（「了解更多」「展开全部 52 阶」） */
@Composable
fun PlanDisclosure(openTitle: String, closeTitle: String, theme: Parchment = Parchment.light, content: @Composable () -> Unit) {
    var open by remember { mutableStateOf(false) }
    Column(Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(12.dp)) {
        Row(Modifier.heightIn(min = 44.dp).clickableNoRipple { open = !open }, verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(if (open) closeTitle else openTitle, color = theme.muted.toColor(), fontSize = 16.sp, fontWeight = FontWeight.SemiBold)
            MaterialIcon(if (open) MI.EXPAND_LESS else MI.EXPAND_MORE, 22f, theme.muted.toColor())
        }
        AnimatedVisibility(open) { content() }
    }
}

/** 跳到另一条路线的卡：标题 17/600 + 一句 14 + 右箭头 */
@Composable
fun PlanLinkCard(title: String, lead: String, theme: Parchment = Parchment.light, onClick: () -> Unit) {
    Row(
        Modifier.fillMaxWidth().clip(RoundedCornerShape(14.dp)).background(theme.surface.toColor().copy(alpha = 0.6f))
            .border(0.5.dp, theme.border.toColor(), RoundedCornerShape(14.dp)).clickableNoRipple(onClick).padding(16.dp),
        verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(title, color = theme.ink.toColor(), fontSize = 17.sp, fontWeight = FontWeight.SemiBold)
            Text(lead, color = theme.muted.toColor(), fontSize = 15.sp)
        }
        MaterialIcon(MI.ARROW_FORWARD, 24f, theme.ink.toColor())
    }
}

File: apps/askbible-ios/AskBible/Read/PlanWidgets.swift
import SwiftUI

/// 读经计划页的手机版视觉件（两端同构，Android 见 ui/PlanWidgets.kt）。
/// Josh 2026-09-09：介绍与内容要为手机看优化 —— 字不小于 13，段落改成要点，能点的做成卡片，整体简短、可视。
/// 文案来自 data/bible-reading-plans/mobile-brief.zh-CN.json（PlanText = PlanCopy 的 mobile.* 键）。
enum PlanText {
    static func t(_ key: String) -> String { PlanCopy.t("mobile." + key) }
    static func f(_ key: String, _ args: [String: String]) -> String { PlanCopy.f("mobile." + key, args) }
}

enum PlanIcons {
    /// mobile-brief 里的图标名 → Material 字形
    static func glyph(_ name: String) -> String {
        switch name {
        case "today": return MI.today
        case "loop": return MI.loop
        case "swap": return MI.swapHoriz
        case "stairs": return MI.stairs
        case "repeat": return MI.repeatGlyph
        case "book": return MI.menuBook
        case "sync": return MI.sync
        case "spa": return MI.spa
        case "replay": return MI.replay
        case "calendar": return MI.calendarMonth
        case "layers": return MI.layers
        default: return MI.today
        }
    }

    /// 轨道图标与色：旧约蓝 / 新约橙 / 智慧书绿（目录页的约别配色）
    static func track(_ id: String) -> (glyph: String, color: Color) {
        switch id {
        case "ot": return (MI.historyEdu, Color(rgb: 0x2E5E8C))
        case "nt": return (MI.autoStories, Color(rgb: 0xB8611E))
        default: return (MI.lightbulb, Color(rgb: 0x3F7A4A))
        }
    }
}

/// 要点胶囊：图标 16 + 文字 14/600
struct PlanFactChip: View {
    let fact: PlanFact
    private let theme = Parchment.light
    var body: some View {
        HStack(spacing: 6) {
            MaterialIcon(glyph: PlanIcons.glyph(fact.icon), size: 18, color: theme.muted)
            Text(fact.text).font(.system(size: 15, weight: .semibold)).foregroundStyle(theme.ink)
        }
        .padding(.horizontal, 12).padding(.vertical, 7)
        .background(Capsule().fill(theme.surface.opacity(0.85)))
        .overlay(Capsule().strokeBorder(theme.border, lineWidth: 0.5))
    }
}

/// 要点 chips 一行，放不下就折行（卡片里三颗 chip 曾被挤得每颗都换行）
struct PlanFactRow: View {
    let facts: [PlanFact]
    var centered = false
    var body: some View {
        PlanFlowLayout(centered: centered) {
            ForEach(Array(facts.enumerated()), id: \.offset) { _, f in PlanFactChip(fact: f).fixedSize() }
        }
        .frame(maxWidth: .infinity, alignment: centered ? .center : .leading)
    }
}

/// 自动换行的横排（iOS 16 Layout）：放不下就折到下一行；centered 时每行居中
struct PlanFlowLayout: Layout {
    var spacing: CGFloat = 8
    var rowSpacing: CGFloat = 8
    var centered = false

    func sizeThatFits(proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) -> CGSize {
        let width = proposal.width ?? .infinity
        let rows = layoutRows(width: width, subviews: subviews)
        let height = rows.reduce(0) { $0 + $1.height } + CGFloat(max(0, rows.count - 1)) * rowSpacing
        let widest = rows.map(\.width).max() ?? 0
        return CGSize(width: width == .infinity ? widest : width, height: height)
    }

    func placeSubviews(in bounds: CGRect, proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) {
        let rows = layoutRows(width: bounds.width, subviews: subviews)
        var y = bounds.minY
        for row in rows {
            var x = bounds.minX + (centered ? max(0, (bounds.width - row.width) / 2) : 0)
            for index in row.indices {
                let size = subviews[index].sizeThatFits(.unspecified)
                subviews[index].place(at: CGPoint(x: x, y: y + (row.height - size.height) / 2), proposal: .unspecified)
                x += size.width + spacing
            }
            y += row.height + rowSpacing
        }
    }

    private struct Row {
        var indices: [Int] = []
        var width: CGFloat = 0
        var height: CGFloat = 0
    }

    private func layoutRows(width: CGFloat, subviews: Subviews) -> [Row] {
        var rows: [Row] = [Row()]
        for (i, sub) in subviews.enumerated() {
            let size = sub.sizeThatFits(.unspecified)
            var row = rows[rows.count - 1]
            let extra = row.indices.isEmpty ? size.width : size.width + spacing
            if !row.indices.isEmpty, row.width + extra > width {
                rows.append(Row(indices: [i], width: size.width, height: size.height))
            } else {
                row.indices.append(i)
                row.width += extra
                row.height = max(row.height, size.height)
                rows[rows.count - 1] = row
            }
        }
        return rows
    }
}

/// 「怎么读」一行：金色圆底图标 + 16/500 短句
struct PlanHowRow: View {
    let fact: PlanFact
    private let theme = Parchment.light
    var body: some View {
        HStack(alignment: .center, spacing: 12) {
            ZStack {
                Circle().fill(Brand.logo.opacity(0.2))
                MaterialIcon(glyph: PlanIcons.glyph(fact.icon), size: 20, color: theme.ink)
            }
            .frame(width: 36, height: 36).contentShape(Rectangle())
            Text(fact.text).font(.system(size: 17, weight: .medium)).lineSpacing(6).foregroundStyle(theme.inkSoft)
            Spacer(minLength: 0)
        }
    }
}

/// 今日读经卡：轨道图标 / 轨道名 13 / 书章 22/700 / 说明 14 / 可选进度条；整卡可点进章
struct PlanTodayCard: View {
    var track: String?
    let label: String
    let title: String
    var subtitle: String?
    var progress: Double?
    let action: () -> Void
    private let theme = Parchment.light
    var body: some View {
        Button(action: action) {
            HStack(spacing: 14) {
                if let track {
                    let style = PlanIcons.track(track)
                    ZStack {
                        Circle().fill(style.color.opacity(0.14))
                        MaterialIcon(glyph: style.glyph, size: 24, color: style.color)
                    }
                    .frame(width: 46, height: 46)
                }
                VStack(alignment: .leading, spacing: 4) {
                    Text(label).font(.system(size: 14, weight: .semibold)).kerning(0.6).foregroundStyle(theme.faint)
                    Text(title).font(.system(size: 22, weight: .bold)).foregroundStyle(theme.ink).lineLimit(2)
                    if let subtitle { Text(subtitle).font(.system(size: 15)).foregroundStyle(theme.muted) }
                    if let progress { PlanProgressBar(value: progress).padding(.top, 4) }
                }
                Spacer(minLength: 0)
                MaterialIcon(glyph: MI.chevronRight, size: 26, color: theme.faint)
            }
            .padding(.horizontal, 16).padding(.vertical, 14)
            .background(RoundedRectangle(cornerRadius: 14).fill(Color(rgb: 0xFFFCF5, opacity: 0.72)))
            .overlay(RoundedRectangle(cornerRadius: 14).strokeBorder(theme.border, lineWidth: 0.5))
        }
        .buttonStyle(.plain)
    }
}

struct PlanProgressBar: View {
    let value: Double
    private let theme = Parchment.light
    var body: some View {
        GeometryReader { g in
            ZStack(alignment: .leading) {
                Capsule().fill(theme.border)
                Capsule().fill(Brand.logo).frame(width: g.size.width * min(1, max(0, value)))
            }
        }
        .frame(height: 5)
    }
}

/// 整宽主按钮 52 高：filled = 墨底米字；否则米底墨字（次要）
struct PlanPrimaryButton: View {
    let title: String
    var filled = true
    let action: () -> Void
    private let theme = Parchment.light
    var body: some View {
        Button(action: action) {
            Text(title).font(.system(size: 17, weight: .semibold))
                .foregroundStyle(filled ? Color(rgb: 0xF5EFE4) : theme.ink)
                .frame(maxWidth: .infinity).frame(height: 52)
                .background(RoundedRectangle(cornerRadius: 14).fill(filled ? theme.ink : theme.surface))
                .overlay(RoundedRectangle(cornerRadius: 14).strokeBorder(filled ? Color.clear : theme.ink.opacity(0.35), lineWidth: 1))
        }
        .buttonStyle(.plain)
    }
}

struct PlanSectionHeader: View {
    let title: String
    var hint: String?
    private let theme = Parchment.light
    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(title).font(.system(size: 20, weight: .bold)).foregroundStyle(theme.ink)
            if let hint { Text(hint).font(.system(size: 15)).foregroundStyle(theme.muted) }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

/// 「✓ 当前计划 · 第 N 天」金色胶囊
struct PlanStatusPill: View {
    let text: String
    var body: some View {
        HStack(spacing: 6) {
            MaterialIcon(glyph: MI.checkCircle, size: 18, color: Color(rgb: 0x8A5A00))
            Text(text).font(.system(size: 15, weight: .semibold)).foregroundStyle(Color(rgb: 0x5B3A00))
        }
        .padding(.horizontal, 12).padding(.vertical, 7)
        .background(Capsule().fill(Brand.logo.opacity(0.26)))
    }
}

/// 单选格（深读节奏 / 起算方式）：标题 18/700 + 副题 13，选中金底金边
struct PlanChoiceTile: View {
    let title: String
    let subtitle: String
    let on: Bool
    let action: () -> Void
    private let theme = Parchment.light
    var body: some View {
        Button(action: action) {
            VStack(spacing: 4) {
                Text(title).font(.system(size: 18, weight: .bold)).foregroundStyle(theme.ink)
                Text(subtitle).font(.system(size: 14, weight: .medium)).foregroundStyle(theme.muted).multilineTextAlignment(.center)
            }
            .frame(maxWidth: .infinity).padding(.vertical, 12).padding(.horizontal, 8)
            .background(RoundedRectangle(cornerRadius: 12).fill(on ? Brand.logo.opacity(0.28) : theme.surface.opacity(0.6)))
            .overlay(RoundedRectangle(cornerRadius: 12).strokeBorder(on ? Brand.logo : theme.border, lineWidth: on ? 1.5 : 0.5))
        }
        .buttonStyle(.plain)
    }
}

/// 第几天步进器：44 触控键 + 24/700 数字
struct PlanStepper: View {
    @Binding var value: Int
    let range: ClosedRange<Int>
    private let theme = Parchment.light
    var body: some View {
        HStack(spacing: 16) {
            step(MI.remove, enabled: value > range.lowerBound) { value = max(range.lowerBound, value - 1) }
            HStack(spacing: 4) {
                Text("第").font(.system(size: 16)).foregroundStyle(theme.muted)
                Text("\(value)").font(.system(size: 24, weight: .bold)).foregroundStyle(theme.ink).frame(minWidth: 44)
                Text("天").font(.system(size: 16)).foregroundStyle(theme.muted)
            }
            step(MI.add, enabled: value < range.upperBound) { value = min(range.upperBound, value + 1) }
        }
    }
    private func step(_ glyph: String, enabled: Bool, action: @escaping () -> Void) -> some View {
        Button(action: action) {
            MaterialIcon(glyph: glyph, size: 22, color: theme.ink)
                .frame(width: 44, height: 44).contentShape(Rectangle())
                .background(RoundedRectangle(cornerRadius: 10).fill(theme.surface))
                .overlay(RoundedRectangle(cornerRadius: 10).strokeBorder(theme.border, lineWidth: 0.5))
                .opacity(enabled ? 1 : 0.4)
        }
        .buttonStyle(.plain).disabled(!enabled)
    }
}

/// 展开 / 收起（「了解更多」「展开全部 52 阶」）
struct PlanDisclosure<Content: View>: View {
    let openTitle: String
    let closeTitle: String
    @ViewBuilder let content: () -> Content
    @State private var open = false
    private let theme = Parchment.light
    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            Button { withAnimation(.easeInOut(duration: 0.2)) { open.toggle() } } label: {
                HStack(spacing: 4) {
                    Text(open ? closeTitle : openTitle).font(.system(size: 16, weight: .semibold)).foregroundStyle(theme.muted)
                    MaterialIcon(glyph: open ? MI.expandLess : MI.expandMore, size: 22, color: theme.muted)
                }
                .frame(minHeight: 44)
            }
            .buttonStyle(.plain)
            if open { content() }
        }
    }
}

/// 跳到另一条路线的卡：标题 17/600 + 一句 14 + 右箭头
struct PlanLinkCard: View {
    let title: String
    let lead: String
    let action: () -> Void
    private let theme = Parchment.light
    var body: some View {
        Button(action: action) {
            HStack(spacing: 12) {
                VStack(alignment: .leading, spacing: 4) {
                    Text(title).font(.system(size: 17, weight: .semibold)).foregroundStyle(theme.ink)
                    Text(lead).font(.system(size: 15)).foregroundStyle(theme.muted)
                }
                Spacer(minLength: 0)
                MaterialIcon(glyph: MI.arrowForward, size: 24, color: theme.ink)
            }
            .padding(16)
            .background(RoundedRectangle(cornerRadius: 14).fill(theme.surface.opacity(0.6)))
            .overlay(RoundedRectangle(cornerRadius: 14).strokeBorder(theme.border, lineWidth: 0.5))
        }
        .buttonStyle(.plain)
    }
}

/// 子页左上返回键（RN ShellSystemBackButton）
/// 左上返回箭头（整块 44×44 可点：图标外的透明部分默认不算点击区，真机点 x<24 没反应）：页面在 GeometryReader（不 ignoresSafeArea）里，内容本来就从安全区下方开始，这里只留 8pt，
/// 不能再加 safeAreaInsets.top —— 那会把箭头压到比系统默认位置低一个状态栏（Josh 真机 2026-09-10）
struct PlanBackButton: View {
    let action: () -> Void
    var body: some View {
        Button(action: action) {
            Image(systemName: "arrow.left")
                .font(.system(size: 22, weight: .medium))
                .foregroundStyle(Parchment.light.ink)
                .frame(width: 44, height: 44).contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .padding(.leading, 12)
        .padding(.top, 8)
    }
}
